src/Io/Transaction.php

```diff
@@ -144,6 +142,29 @@ private function onResponseRedirect(ResponseInterface $response, RequestInterfac
         return $this->next($request);
     }
 
+    /**
+     * @param RequestInterface $request
+     * @param UriInterface $location
+     * @return \Clue\React\Buzz\Message\RequestInterface
+     */
+    private function makeRedirectRequest(RequestInterface $request, UriInterface $location)
+    {
+        $originalHost = $request->getUri()->getHost();
+        $request = $request
+            ->withoutHeader('Host')
+            ->withoutHeader('Content-Type')
+            ->withoutHeader('Content-Length');
+
+        if($location->getHost() !== $originalHost) {
```

Comment:
It's my understanding this should probably verify the whole authority, no? (For example when redirecting from https:// to http:// or using another port etc.)

Also, this block could use a small comment (*why*), plus minor CS fix :+1:

Reply:
We remove authorization only on changing hostname  (not if just changing ports or protocols). I've added a description comment and fixed CS.